### PR TITLE
bump bevy_tasks futures-lite to 2.0.1

### PR DESCRIPTION
**Pull request by tbillington**
# Objective

Updates [`futures-lite`](https://github.com/smol-rs/futures-lite) in bevy_tasks to the next major version (1 -> 2).

Also removes the duplication of `futures-lite`, as `async-fs` requires v 2, so there are currently 2 copies of futures-lite in the dependency tree.

Futures-lite has received [a number of updates](https://github.com/smol-rs/futures-lite/blob/master/CHANGELOG.md) since bevy's current version `1.4`.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by hymm** [comment]
looking at `cargo tree` there are still other versions of `futures-lite < 2` in tree. One in bevy_assets and one in dev-dependencies. I also think we'd need to upgrade async-fs to v2 to remove futures lite 1. We should at least do the explicit dependencies in bevy_assets and the root Cargo.toml in this pr.

**Comment by tbillington** [comment]
Good catch @hymm , I think I accidentally ran cargo tree in my project repo instead of bevy so I missed those.

I will re-investigate after https://github.com/bevyengine/bevy/pull/10370 is merged :)

**Comment by alice-i-cecile** [comment]
Merging the linked PR for you now :)

**Comment by tbillington** [comment]
I've updated all usages of `futures-lite` and `async-executor` which was using v1. Running cargo tree now only gives me `futures-lite` 2.0.1.

File: Cargo.toml [hunk 1]
```diff
@@ -294,7 +294,7 @@ flate2 = "1.0"
 serde = { version = "1", features = ["derive"] }
 bytemuck = "1.7"
 # Needed to poll Task examples
-futures-lite = "1.11.3"
+futures-lite = { version = "2.0.1" }
```
Comment (hymm) [inline Cargo.toml:297]: should probably remove the `version` for consistency.
 
```suggestion
futures-lite = "2.0.1"
```

File: crates/bevy_asset/Cargo.toml [hunk 2]
```diff
@@ -32,7 +32,7 @@ async-lock = "3.0"
 crossbeam-channel = "0.5"
 downcast-rs = "1.2"
 futures-io = "0.3"
-futures-lite = "1.12"
+futures-lite = { version = "2.0.1" }
```
Comment (alice-i-cecile) [inline crates/bevy_asset/Cargo.toml:35]:
```suggestion
futures-lite = "2.0.1" 
```

File: crates/bevy_render/Cargo.toml [hunk 3]
```diff
@@ -75,7 +75,7 @@ smallvec = { version = "1.6", features = ["union", "const_generics"] }
 downcast-rs = "1.2.0"
 thread_local = "1.1"
 thiserror = "1.0"
-futures-lite = "1.4.0"
+futures-lite = { version = "2.0.1" }
```
Comment (alice-i-cecile) [inline crates/bevy_render/Cargo.toml:78]:
```suggestion
futures-lite = "2.0.1" 
```

File: crates/bevy_tasks/Cargo.toml [hunk 4]
```diff
@@ -12,8 +12,8 @@ keywords = ["bevy"]
 multi-threaded = []
 
 [dependencies]
-futures-lite = "1.4.0"
-async-executor = "1.3.0"
+futures-lite = { version = "2.0.1" }
```
Comment (alice-i-cecile) [inline crates/bevy_tasks/Cargo.toml:15]:
```suggestion
futures-lite = "2.0.1"
```

**Comment by alice-i-cecile** [comment]
@tbillington I resolved the formatting issue for you :)

**Review by hymm — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request merged**